bmc/firmware.go

```diff
@@ -165,3 +169,294 @@ func FirmwareInstallStatusFromInterfaces(ctx context.Context, installVersion, co
 
 	return firmwareInstallStatus(ctx, installVersion, component, taskID, implementations)
 }
+
+// FirmwareInstallerWithOpts defines an interface to install firmware that was previously uploaded with FirmwareUpload
+type FirmwareInstallerUploaded interface {
+	// FirmwareInstallUploaded uploads firmware update payload to the BMC returning the firmware install task ID
+	//
+	// parameters:
+	// component - the component slug for the component update being installed.
+	// uploadTaskID - the taskID for the firmware upload verify task (returned by FirmwareUpload)
+	//
+	// return values:
+	// installTaskID - A installTaskID is returned if the update process on the BMC returns an identifier for the firmware install process.
+	FirmwareInstallUploaded(ctx context.Context, component, uploadTaskID string) (taskID string, err error)
+}
+
+// firmwareInstallerProvider is an internal struct to correlate an implementation/provider and its name
+type firmwareInstallerWithOptionsProvider struct {
+	name string
+	FirmwareInstallerUploaded
+}
+
+// firmwareInstallUploaded uploads and initiates firmware update for the component
+func firmwareInstallUploaded(ctx context.Context, component, uploadTaskID string, generic []firmwareInstallerWithOptionsProvider) (installTaskID string, metadata Metadata, err error) {
```

Comment:
In general I think named returns are a maintenance hazard.

Reply:
I kinda agree, although I have them there since they are indicative of what the returned string holds.